timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -272,12 +272,18 @@ export class TimeGraphChart extends TimeGraphChartLayer {
                 this.mouseEndX = e.offsetX;
                 const start = this.mouseZoomingStart;
                 const end = this.unitController.viewRange.start + (this.mouseEndX / this.stateController.zoomFactor);
-                if (start !== end) {
-                    this.unitController.viewRange = {
-                        start: Math.max(Math.min(start, end), this.unitController.viewRange.start),
-                        end: Math.min(Math.max(start, end), this.unitController.viewRange.end)
-                    }
+                
+                let newViewStart = Math.max(Math.min(start, end), this.unitController.viewRange.start);
+                let newViewEnd = Math.min(Math.max(start, end), this.unitController.viewRange.end);
+                if (newViewEnd - newViewStart < 1) {
+                    newViewStart = Math.trunc(Math.min(newViewStart, newViewEnd));
```

Comment:
I think from above newViewStart is guaranteed to be the smallest of the two right? So the min is unnecessary.

I would consider adding trunc() on newViewStart and ceil() on newViewEnd above ('let' statements), so that they are always integers, what do you think?

Come to think of it, would that make the < 1 check unnecessary? Since if start and end are different, their trunc() and ceil() cannot possibly be the same integer?